realm/src/main/java/io/realm/Realm.java

```diff
@@ -813,17 +809,21 @@ private static void initializeRealm(Realm realm) {
         }
         checkHasPrimaryKey(clazz);
 
-        JsonReader reader = new JsonReader(new InputStreamReader(in, "UTF-8"));
+        // As we need the primary key value we have to first parse the entire input stream as in the general
```

Comment:
We shouldn't have a method just for symmetry reasons. If it is not useful, please remove it.